app/docs/enterprise/0.31-x/oidc-auth0.md

```diff
@@ -0,0 +1,41 @@
+---
+title: OpenID Connect with Auth0
+class: page-install-method
+---
+
+This guide covers an example OpenID Connect plugin configuration to authenticate headless services using Auth0's identity provider.
+
+# Auth0 IDP configuration
+
+This configuration will use a [client credentials grant](https://auth0.com/docs/api-auth/tutorials/client-credentials) as it is non-interactive, and because we expect clients to authenticate on behalf of themselves, not an end-user. To do so, you will need to [create an Auth0 API](https://auth0.com/docs/apis#how-to-configure-an-api-in-auth0) and a [non-interactive client](https://auth0.com/docs/clients).
+
+## API configuration
+
+When creating your API, you will need to specify an Identifier. Using the URL that your service makes requests to is generally appropriate, so this will typically be the hostname/path combination configured as a service or API in Kong.
+
+After creating your API, you will also need to add the `openid` scope at `https://manage.auth0.com/#/apis/<API ID>/scopes`.
+
+## Client configuration
+
+You will need to authorize your client to access your API. Auth0 will prompt you to do so after client creation if you select the API you created previously from the client's Quick Start menu. After toggling the client to Authorized, expand its authorization settings and enable the `openid` scope.
+
+# Kong configuration
+
+If you have not done so already, [create a service](https://getkong.org/docs/0.13.x/admin-api/#add-service) to protect. The `url` configuration should match the Identifier you used when configuring Auth0.
+
+Add an OpenID plugin configuration following the example below. Auth0's token endpoint [requires passing the API identifier in the `audience` parameter](https://auth0.com/docs/api/authentication#client-credentials), which must be added as a custom argument:
+
+```bash
+$ curl -i -X POST http://kong:8001/kong-admin/services/<service name>/plugins --data name="openid-connect" \
+--data config.auth_methods="client_credentials" \
+--data config.issuer="https://<auth0 API name>.auth0.com/.well-known/openid-configuration" \
+--data config.token_post_args_names="audience" \
+--data config.token_post_args_values="https://example.com/"
+```
+
+# Downstream configuration
+
+The service accessing your resource will need to pass its credentials to Kong. It can do so via HTTP basic authentication, query string arguments, or parameters in the request body. Basic authentication is generally preferable, as credentials in a query string will be present in Kong's access logs (and possibly in other infrastructure's access logs, if you have HTTP-aware infrastracture in front of Kong) and credentials in request bodies are limited to request methods that expect
```

Comment:
When you say "the service" do you mean "the client"? Typically we reserve the word "service" to refer to the services in "services and routes"